python/ray/dag/compiled_dag_node.py

```diff
@@ -309,7 +326,17 @@ def _get_or_compile(
                 self.dag_output_channels,
             )
 
-        queue = [self.input_task_idx]
+        # somehow the input_attribute_nodes attr of InputNode is lost,
+        # reconstruct it here
+        input_node = None
+        for dag_node in self.dag_node_to_idx.keys():
+            if isinstance(dag_node, InputNode):
+                input_node = dag_node
+        for dag_node in self.dag_node_to_idx.keys():
+            if isinstance(dag_node, InputAttributeNode):
+                input_node.input_attribute_nodes[dag_node._key] = dag_node
```

Comment:
Where do we use this? It seems to be set here but I didn't see if it gets read anywhere in the PR.

Reply:
only used in an earlier version, not useful now, removed